src/main/java/org/example/learningsystem/email/config/EmailServerProperties.java

```diff
@@ -12,11 +15,32 @@
 @Validated
 @Getter
 @Setter
+@AllArgsConstructor
+@NoArgsConstructor
 public class EmailServerProperties {
 
     @NotNull
+    @DestinationProperty("mail.smtp.from")
     private String from;
-    private String name;
     @NotNull
-    private String token;
-}
\ No newline at end of file
+    @DestinationProperty("mail.user")
+    private String user;
+    @NotNull
+    @DestinationProperty("mail.password")
+    private String password;
+    @NotNull
+    @DestinationProperty("mail.smtp.host")
```

Comment:
idea is really cool, but it could be much easier just to map required properties manually :)